internal/pkg/utils/url/url.go

```diff
@@ -23,30 +25,93 @@ func ParseQuery(query string) (m *orderedmap.OrderedMap, err error) {
 			continue
 		}
 		key, value, _ := strings.Cut(key, "=")
-		key, err1 := url.QueryUnescape(key)
-		if err1 != nil {
-			if err == nil {
-				err = err1
+		key, err = url.QueryUnescape(key)
+		if err != nil {
+			return m, err
+		}
+		value, err = url.QueryUnescape(value)
+		if err != nil {
+			return m, err
+		}
+
+		path, sliceAppend, err := parseKey(key)
```

Comment:
Another `n` complexity. Isn't it more efficient to replace the `unescaped` character right away instead of `unescaping` it and then parsing it again?